### PR TITLE
Allow delist for negative transfer value

### PR DESCRIPTION
**Pull request by Erebos0506**
Allowed buy and delist interactions after nft becomes untransferable to prevent the nft from being stuck as listed

### CODE REVIEW

File: src/rmrk2.0.0/tools/consolidator/utils.ts [hunk 1]
```diff
@@ -7,25 +7,35 @@ import {
...
 import { Remark } from "./remark";
 import { OP_TYPES } from "../constants";
 
-export const isNftTransferable = (nft: NFT, remark: Remark) => {
-  return (
-    nft.transferable === 1 ||
-    (nft.transferable < 0 && nft.block - nft.transferable >= remark.block) ||
-    (nft.transferable > 1 && remark.block >= nft.transferable)
-  );
+export const isNftTransferable = (nft: NFT, remark: Remark, opType: OP_TYPES) => {
```
Comment (ilionic) [inline src/rmrk2.0.0/tools/consolidator/utils.ts:16]: * please add some docs comments about extra logic and unit tests.
* prefer strict equality / non equality
* multiple `else` conditions could be omitted

**Review by ilionic — CHANGES_REQUESTED** [review]

File: src/rmrk2.0.0/tools/consolidator/utils.ts (hunk 1 above)
Comment (Erebos0506) [reply src/rmrk2.0.0/tools/consolidator/utils.ts:16]: Added comments and changed to strict equal but by removing else conditions i'd change the logic and remove the changes i made

Comment (Erebos0506) [reply src/rmrk2.0.0/tools/consolidator/utils.ts:16]: You were right i removed one of the else conditions

File: src/rmrk2.0.0/tools/consolidator/utils.ts [hunk 2]
```diff
@@ -7,25 +7,37 @@ import {
...
+  //check transferability if not buy or delist interactions
+  if (opType === "LIST" && List.fromRemark(remark.remark).price !== 0 || opType !== "BUY") {
```
Comment (Yuripetusko) [inline src/rmrk2.0.0/tools/consolidator/utils.ts:18]: I would instead move this logic to list interaction. 

```
if (listEntity.price !== BigInt(0)) {
  validateTransferability(nft, remark, OP_TYPES.LIST);
}
```

Comment (Erebos0506) [reply src/rmrk2.0.0/tools/consolidator/utils.ts:18]: Should i do the same for buy or would you rather have the nfts not be buyable after the transferable block passed?
```
if (nft.transferable >= 0) {
  validateTransferability(nft, remark, OP_TYPES.BUY);
}
```

Comment (Yuripetusko) [reply src/rmrk2.0.0/tools/consolidator/utils.ts]: Not buyable IMO

Comment (Erebos0506) [reply src/rmrk2.0.0/tools/consolidator/utils.ts]: Wouldn't that make it seem like a scam with it being listed but not buyable?

Comment (Erebos0506) [reply src/rmrk2.0.0/tools/consolidator/utils.ts]: A workaround might be delisting it on failed buy if listed after the blocknumber passed but i'm not sure how good it is that this'd require a failed buy interaction

Comment (Yuripetusko) [reply src/rmrk2.0.0/tools/consolidator/utils.ts]: nah I think we can just handle it on UI if really needed

Comment (Erebos0506) [reply src/rmrk2.0.0/tools/consolidator/utils.ts]: Ok

**Comment by Yuripetusko** [comment]
Update test snapshot please

**Comment by Erebos0506** [comment]
> Update test snapshot please

wip ;)

**Comment by Erebos0506** [comment]
Done

**Comment by Yuripetusko** [comment]
Thank you!

**Pull request merged**